tests/src/test/scala/cats/tests/CoproductTests.scala

```diff
@@ -0,0 +1,48 @@
+package cats.tests
+
+import algebra.Eq
+import algebra.laws.OrderLaws
+import cats._
+import cats.data.Coproduct
+import cats.functor.Contravariant
+import cats.laws.discipline._
+import cats.laws.discipline.arbitrary._
+import org.scalacheck.Arbitrary
+
+class CoproductTests extends CatsSuite {
+
+  checkAll("Coproduct[Option, Option, ?]", TraverseTests[Coproduct[Option, Option, ?]].traverse[Int, Int, Int, Int, Option, Option])
+  checkAll("Traverse[Coproduct[Option, Option, ?]]", SerializableTests.serializable(Traverse[Coproduct[Option, Option, ?]]))
+
+  checkAll("Coproduct[Option, Option, ?]", FoldableTests[Coproduct[Option, Option, ?]].foldable[Int, Int])
```

Comment:
Unfortunately, I think this will pick up the same instances as the `.traverse` check above. That's why I recommended the somewhat bizarre `ListWrapper.foldable` approach in another comment.